src/main/java/com/example/demo/service/ItemService.java

```diff
@@ -3,24 +3,30 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import com.example.demo.entity.Item;
+import com.example.demo.exception.ResourceNotFoundException;
 import com.example.demo.repository.ItemMapper;
 
 @Service
 public class ItemService {
+	private final ItemMapper itemMapper;
 
-	@Autowired
-	ItemMapper itemMapper;
+	public ItemService(ItemMapper itemMapper) {
+		this.itemMapper = itemMapper;
+	}
 
 	// 1件検索
-	public Item findById(Integer id) {
-		Item item = new Item();
-		item.setId(id);
-		return itemMapper.findById(item);
+	public Optional<Item> findById(Integer id) {
+		try {
+			Optional<Item> optionalId = itemMapper.findById(id);
+			return optionalId;
+		} catch (ResourceNotFoundException e) {
+			return null;
+		}
 	}
```

Comment:
こちらOptional<Item>を返すのであれば
```suggestion
	public Optional<Item> findById(Integer id) {
		return itemMapper.findById(id);
	}
```
でよいです。

もしExceptionを返すのであれば
```suggestion
	public Item findById(Integer id) {
		Optional<Item> item = itemMapper.findById(id);
		return item.orElseThrow(ResourceNotFoundException::new);
	}
```
のようにServiceで例外をthrowしてControllerでキャッチするのが適切かと思います。
（このあたりの実装ルールはプロジェクトの方針にもよります）